### DISCUSSION TITLE
How does SF select initial candidate moves?

### DISCUSSION THREAD

**Discussion by Jack4264** [official-stockfish/Stockfish#4134] [Q&A]
Hi, first time posting here. I'm a chess player not a programmer. 

In running Deep Position Analysis in the Fritz interface using SF 15, sometimes lower down the tree with few branching like 2, different runs produce different candidate nodes. The second run is slightly longer than the first and has a couple more nodes at the first branching point. I don't think these differences affect the 4th and last branching point. 

So it appears candidate node selection is somewhat random? Or with minimal resources in evaluating and choosing candidates? These nodes happen to be very close in evaluation. 

I went through an online youtube tutorial on chess engine processes. I understand most of it but the initial candidate selection was never clear. Can someone please define this in a clear manner? 

As a chess player, I need to understand the optimal setting for the calculation for the exercise. Thank you.

**Comment by vondele** [comment]
I think your question is not quite clear. If you use multiple threads, the found best moves and principal variations could be different every time you search, i.e. multi-threaded search is not deterministic. If you would want results to be exactly deterministic, you need to use a single thread, and GUI has to initialize the engine to a clean state (send the command `ucinewgame`), but single threaded search is obviously weaker than multithreaded search. 

Optimal settings are quite straightforward, ... use the latest version of SF, with as many threads and as much hash as possible, and if you have table bases on fast storage (SSD), you can use those as well. The longer you let the engine search the more accurate the answer will be.
